chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/CertifiedProductDownloadableResourceCreatorJob.java

```diff
@@ -126,6 +130,7 @@ private void initializeWritingToFiles(CertifiedProductXmlPresenter xmlPresenter,
         xmlPresenter.open(tempXmlFile);
 
         csvPresenter.setLogger(LOGGER);
+        csvPresenter.setFf4j(ff4j);
```

Comment:
This might be a bad a pattern I introduced when this was originally written, but we should fix it.  By having these 'setters' it is possible to call `csvPresenter.open( ... )` without having passed in the `Ff4j` or the `LOGGER`.  
Do you think we should make these parameters in the `.open( ... )` method?  That way we are sure that all dependencies exist when making the call?

Reply:
The only `csvPresenter.open()` call we have is looking for a File input: 
![image](https://user-images.githubusercontent.com/909012/103648903-92cc9080-4f2b-11eb-89f5-d23fd227438d.png)

I'm not sure if it was a bad pattern when you introduced it, but it was you that did it, though it's also now been this way since Oct of 2018: 
![image](https://user-images.githubusercontent.com/909012/103648979-b263b900-4f2b-11eb-9751-204f9fa1d512.png)

I just opened up the code to see if I could switch to the requested change, but I'm not sure how I'd go about doing that. If you'd like to pull this branch to your fork, make whatever changes would be necessary, and PR it back to this branch I'd love to see what you're talking about. Would you also want to make these changes in the XML presenter?